src/hooks/useAccountBalance.ts

```diff
@@ -52,13 +50,38 @@ export const useAccountBalance = ({
   const evmChainId = Number(useEnvConfig('ethereumChainId'));
   const stakingBalances = useAppSelector(getStakingBalances, shallowEqual);
 
-  const evmQuery = useBalance({
-    enabled: Boolean(!isCosmosChain && addressOrDenom?.startsWith('0x')),
+  const isEVMnativeToken = addressOrDenom === CHAIN_DEFAULT_TOKEN_ADDRESS;
+
+  const evmNative = useBalance({
     address: evmAddress,
     chainId: typeof chainId === 'number' ? chainId : Number(evmChainId),
-    token:
-      addressOrDenom === CHAIN_DEFAULT_TOKEN_ADDRESS ? undefined : (addressOrDenom as EvmAddress),
-    watch: true,
+    query: {
+      enabled: Boolean(!isCosmosChain && isEVMnativeToken),
+    },
+  });
+
+  const tokenContract = {
```

Comment:
Wagmi 2 handles token balance fetching completely differently. useBalance with token is deprecated.

Instead we have to fetch token via useReadContracts and calling the ABI functions. So we need to split fetching for native token and erc20 tokens separately